controller/internal/processmon/processmon_test.go

```diff
@@ -111,6 +111,26 @@ func TestLaunchProcess(t *testing.T) {
 		p, ok := pm.(*RemoteMonitor)
 		So(ok, ShouldBeTrue)
 
+		// cleanup the errChannel
+		defer func() {
+			// t.Log("Closing the errChannel\n")
+			timeout := make(chan bool, 1)
+			go func() {
+				time.Sleep(1 * time.Second)
+				timeout <- true
+			}()
+			for {
+				select {
+				case <-errChannel:
+					break
+				case <-timeout:
```

Comment:
this can be done more easily with a `<-time.After()`, no?

Reply:
oddly, I tried that first, as it's the most common pattern, but it ended up never reading from the errChannel before closing it, and so it never drained the errChannel and always causes the data race still.  This pattern I tried instead seemed to work better at given the other side time to write to the errChannel and this side time to read from it, and to avoid the data race here.

Reply:
changed to `<-time.After()` and works ok 👍